proto/interchain_security/ccv/provider/v1/provider.proto

```diff
@@ -329,6 +332,11 @@ message ConsumerValidator {
   int64 power = 2;
   // public key the validator uses on the consumer chain during this epoch
   tendermint.crypto.PublicKey consumer_public_key = 3;
+  // height the validator had when it FIRST became a consumer validator
+  // If a validator becomes a consumer validator at height `H` and is continuously a consumer validator for all the upcoming
+  // epochs, then the height of the validator SHOULD remain `H`. This height only resets to a different height if a validator
+  // stops being a consumer validator during an epoch and later becomes again a consumer validator.
+  int64 height = 4;
```

Comment:
Doesn't this mean that we need to migrate all the consumer validator state?

Reply:
I do not think a migration is _necessary_.  The `height` would be set to `0`. So, validators that were consumer validators during the upgrade would be getting rewards without necessarily having validated the consumer chain for more than `NumberOfEpochsToStartReceivingRewards`. But this does not seem that problematic to me because validators that do not validate would stop getting rewards.

The other option, as you suggest, would be to indeed migrate the state with something like this:
```
for _, chainID := range providerKeeper.GetAllRegisteredConsumerChainIDs(ctx) {
  for _, val := range providerKeeper.GetConsumerValSet(ctx, chainID) {
	  val.Height = X ???
	  providerKeeper.SetConsumerValidator(ctx, chainID, val)
  }
}
```
If `X = ctx.BlockHeight()`, then validators that have been validating for more than `NumberOfEpochsToStartReceivingRewards` might not receive any rewards at the beginning. If `X - NumberOfEpochsToStartReceivingRewards * BlocksPerEpoch`, then we might give rewards to validators that have not been validating for more than `NumberOfEpochsToStartReceivingRewards`.

So, it seems to me that a migration is unnecessary. Let me know if you have any concerns.